accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -205,9 +222,17 @@ is declared `late` and does not have an initializer.
 It is an error if the object being iterated over by a `for-in` loop has a static
 type which is not `dynamic`, and is not a subtype of `Iterable<dynamic>`.
 
+It is an error if the type of the value returned from a factory constructor is
+not a subtype of the class type associated with the class in which it is defined
+(specifically, it is an error to return a nullable type from a factory
+constructor for any class other than `Null`).
+
 It is a warning to use a null aware operator (`?.`, `?..`, `??`, `??=`, or
 `...?`) on a non-nullable receiver.
 
+It is a warning to use the null check operator (`!`) on a non-nullable
```

Comment:
Just a warning, not an error?

Reply:
It feels unfriendly to code generators that want to be blind about nullability to make it an error.  Also makes it more breaking to change a nullable return type to a non-nullable return type, etc.